module/behaviour/skills/HeadBehaviourSoccer/src/HeadBehaviourSoccer.cpp

```diff
@@ -58,75 +61,63 @@ namespace module::behaviour::skills {
                 // Create vector of search positions
                 for (const auto& position : config["positions"].config) {
-                    search_positions.push_back(position.as<Expression>());
+                    cfg.search_positions.push_back(position.as<Expression>());
                 }
             });
 
-
-        // TODO(BehaviourTeam): remove this horrible code
         // Check to see if we are currently in the process of getting up.
-        on<Trigger<ExecuteGetup>>().then([this] { isGettingUp = true; });
+        on<Trigger<ExecuteGetup>>().then([this] { is_getting_up = true; });
 
         // Check to see if we have finished getting up.
-        on<Trigger<KillGetup>>().then([this] { isGettingUp = false; });
-
-        // Updates the last seen time of ball
-        on<Trigger<VisionBalls>>().then([this](const VisionBalls& balls) {
-            if (!balls.balls.empty()) {
-                ballLastMeasured = NUClear::clock::now();
-                rBCc             = reciprocalSphericalToCartesian(balls.balls[0].measurements[0].srBCc.cast<double>());
-            }
-        });
+        on<Trigger<KillGetup>>().then([this] { is_getting_up = false; });
 
-        on<Every<90, Per<std::chrono::seconds>>, Sync<HeadBehaviourSoccer>>().then(
+        on<Every<90, Per<std::chrono::seconds>>, Optional<With<FilteredBall>>, Sync<HeadBehaviourSoccer>>().then(
             "Head Behaviour Main Loop",
-            [this]() {
-                // Get the time since the ball was last seen
-                float timeSinceBallLastMeasured =
-                    std::chrono::duration_cast<std::chrono::duration<float>>(NUClear::clock::now() - ballLastMeasured)
-                        .count();
+            [this](const std::shared_ptr<const FilteredBall>& ball) {
                 // Only look for ball if not getting up
-                if (!isGettingUp) {
-                    if (timeSinceBallLastMeasured < search_timeout_ms / 1000) {
+                if (!is_getting_up) {
+                    if (ball && NUClear::clock::now() - ball->time_of_measurement < cfg.search_timeout) {
                         // We can see the ball, lets look at it
-                        Eigen::Vector2d angles               = screenAngularFromObjectDirection(rBCc);
+                        Eigen::Vector3d rBCc   = reciprocalSphericalToCartesian(ball->srBCc.cast<double>());
+                        Eigen::Vector2d angles = screenAngularFromObjectDirection(rBCc);
                         std::unique_ptr<HeadCommand> command = std::make_unique<HeadCommand>();
                         command->yaw                         = angles[0];
-                        command->pitch                       = angles[1];
+                        command->pitch                       = angles[1] + cfg.pitch_offset;
                         command->robot_space                 = true;
                         command->smooth                      = true;
```

Comment:
The pitch offset?

Reply:
no, smooth

Reply:
The smooth flag toggles on a filter in the head controller to smooth angles sent to servos, preventing erratic motion if it's sent noisy commands. It's disabled for the search positions because we don't want to filter those commands. We could change the name to something better.